fenix/app/src/main/java/org/mozilla/fenix/shopping/store/ReviewQualityCheckState.kt

```diff
@@ -89,6 +90,13 @@ sealed interface ReviewQualityCheckState : State {
                 val highlightsFadeVisible: Boolean =
                     highlights != null && showMoreButtonVisible &&
                         highlights.forCompactMode().entries.first().value.size > 1
+
+                /**
+                 * The status of the product analysis.
+                 */
+                enum class AnalysisStatus {
+                    NEEDS_ANALYSIS, REANALYZING, UP_TO_DATE, COMPLETED
```

Comment:
Where are these statuses from? Should they follow the [Fakespot API Docs](https://docs.google.com/document/d/1eppEnQKrSKavM3qmlm36WG7y8o2ggm-oxugnoSr2tpw/edit#heading=h.84nyloh6qhar): `pending, in_progress, completed, not_analyzable, not_found, unprocessable`? Or is this referring to something else?

Reply:
Or are the ones I'm referring to going to be translated to those error states?

Reply:
This is the UI state, only one of these 4 state is possible when analysis is present. The data (api response) layer is separated from UI so all the logic is abstracted in the state/store and not in the UI. 

`pending`, `in_progress` map to `REANALYZING`, the middleware will set it when the api calls successfully return with those statuses.
`completed` maps one to one.
The rest will map to different Error state(s), based on the status response or the logic, e.g. `notfound` has a specific logic based on review grade and other fields.